src/Lumi/Components2/Banner.purs

```diff
@@ -0,0 +1,223 @@
+import Lumi.Styles.Box (FlexAlign(..), _align, _alignSelf, _column, _flex, _interactive, _justify, _row) as S
+import Lumi.Styles.Responsive (desktopQuery)
+import Lumi.Styles.Responsive (onDesktop) as S
+import Lumi.Styles.Theme (LumiTheme(..), useTheme)
+import React.Basic (JSX)
+import React.Basic.DOM as R
+import React.Basic.Emotion as E
+import React.Basic.DOM.Events (capture_)
+import React.Basic.Hooks (useState)
+import React.Basic.Hooks as React
+
+data Banner = Banner
+
+type BannerProps =
+  ( component :: Banner
+  , dismissable :: Boolean
+  , icon :: Maybe JSX
+  , title :: Maybe JSX
+  , content :: Array JSX
+  )
+
+banner :: LumiComponent BannerProps
+banner =
+  unsafePerformEffect do
+    lumiComponent "Banner" defaults \props -> React.do
+      theme <- useTheme
+      visible /\ setVisible <- useState true
+      pure
+        $ Monoid.guard visible
+        $ box
+        $ S.banner
+        $ style props.css
+        $ _ { className = props.className
+            , content =
+                [ box
+                  $ S._column
+                  $ S._flex
+                  $ S._alignSelf S.Center
+                  $ S._align S.Stretch
+                  $ _ { content =
+                          [ case props.title of
+                              Just title ->
+                                box
+                                $ S._alignSelf S.Start
+                                $ style_
+                                    ( css
+                                      { marginBottom: int 8
+                                      }
+                                    )
+                                $ _ { content = [ title ]
+                                    }
+                              _ -> mempty
+                          , box
+                            $ S._align S.Center
+                            $ S._row
+                            $ S._flex
+                            $ _ { content =
+                                    [ case props.icon of
+                                        Just icon ->
+                                          box
+                                          $ S._alignSelf S.Start
+                                          $ S.onDesktop (S._alignSelf S.Center)
+                                          $ style_
+                                              ( css
+                                                  { marginRight: int 16
+                                                  }
+                                              )
+                                          $ _ { content = [ icon ]
+                                              }
+                                        _ -> mempty
+                                    ]
+                                    <> props.content
+                                }
+                          ]
+                      }
+                , Monoid.guard props.dismissable
+                    $ button
```

Comment:
I think we should use an actual `R.button` element here, instead of our `button` component, as style changes to it might affect how it gets displayed here, not to mention it's easier to override the default styling of the HTML `<button>`.

Reply:
This should also help fix the large margin the dismiss button has and the height difference between banners with and without the dismiss button:
![Screen Shot 2020-05-12 at 10 00 17](https://user-images.githubusercontent.com/2164548/81695208-b260f700-9438-11ea-8782-6a08d75e88d0.png)

![Screen Shot 2020-05-12 at 10 04 04](https://user-images.githubusercontent.com/2164548/81694684-01f2f300-9438-11ea-8637-07997792dd9b.png)

Reply:
Yeah agreed, will update.